source/core/smart-pointer.h

```diff
@@ -35,6 +35,11 @@ namespace Slang
             referenceCount++;
         }
 
+        void decreaseReference()
+        {
```

Comment:
What is the point of `decreaseReference` as a distinct operation from `releaseReference`? Is it that it doesn't actually delete the object when the refcount goes to zero?